tests/integration/targets/lambda_layer/tasks/main.yml

```
@@ -0,0 +1,222 @@
+        license_info: GPL-3.0-only
+      register: first_version
+
+    - name: Create another lambda layer version
+      lambda_layer:
+        name: "{{ layer_name }}"
+        description: '{{ resource_prefix }} lambda layer second version'
+        content:
+          s3_bucket: "{{ s3_bucket_name }}"
+          s3_key: "{{ s3_bucket_object }}"
+        compatible_runtimes:
+          - python3.7
+        license_info: GPL-3.0-only
+      register: last_version
+
+    - name: Retrieve all layers with latest version
+      lambda_layer_info:
+      register: layers
+    
+    - name: Ensure layer created above was found
+      assert:
+        that:
+          - '"layers_versions" in layers'
+          - last_version.layer_version.layer_arn in layers_arns
+          - last_version.layer_version.layer_version_arn in layers_version_arns
+          - first_version.layer_version.layer_version_arn not in layers_version_arns
+      vars:
+        layers_arns: '{{ layers.layers_versions | map(attribute="layer_arn") | list }}'
+        layers_version_arns: '{{ layers.layers_versions | map(attribute="layer_version_arn") | list }}'
+
+    - name: Retrieve all layers versions
+      lambda_layer_info:
+        name: "{{ layer_name }}"
+      register: layers
+
+    - name: Ensure layer created above was found
+      assert:
+        that:
+          - '"layers_versions" in layers'
+          - layers.layers_versions | length == 2
+          - last_version.layer_version.layer_version_arn in layers_version_arns
+          - first_version.layer_version.layer_version_arn in layers_version_arns
+      vars:
+        layers_version_arns: '{{ layers.layers_versions | map(attribute="layer_version_arn") | list }}'
+
+    - name: Delete latest layer version
+      lambda_layer:
+        name: "{{ layer_name }}"
+        version: "{{ last_version.layer_version.version }}"
+        state: absent
+      check_mode: true
+      register: delete_check_mode
+
+    - name: Retrieve all layers versions
+      lambda_layer_info:
+        name: "{{ layer_name }}"
+      register: layers
+
+    - name: Ensure layer still contains the same number of versions
+      assert:
+        that:
+          - delete_check_mode is changed
+          - layers.layers_versions | length == 2
+          - last_version.layer_version.layer_version_arn in layers_version_arns
+          - first_version.layer_version.layer_version_arn in layers_version_arns
+      vars:
+        layers_version_arns: '{{ layers.layers_versions | map(attribute="layer_version_arn") | list }}'
+
+    - name: Delete latest layer version
+      lambda_layer:
+        name: "{{ layer_name }}"
+        version: "{{ last_version.layer_version.version }}"
+        state: absent
+      register: delete_layer
+
+    - name: Retrieve all layers versions
+      lambda_layer_info:
+        name: "{{ layer_name }}"
+      register: layers
+
+    - name: Ensure layer still contains the same number of versions
```

```suggestion
    - name: Ensure layer was deleted
```
Nit, the task name was copied from the previous one